topics/user_interface_components/editor_components.md

```diff
@@ -20,6 +20,23 @@ Several commonly needed customization implementations exist, including:
 - [`HorizontalScrollBarEditorCustomization`](upsource:///platform/platform-impl/src/com/intellij/ui/HorizontalScrollBarEditorCustomization.java) to turn on/off horizontal scrollbar
 - [`ErrorStripeEditorCustomization`](upsource:///platform/platform-impl/src/com/intellij/ui/ErrorStripeEditorCustomization.java) to turn on/off error stripes on right
 
+`EditorTextField` has a number of subclasses that can be used as needed for additional features.
+
+If you want to use an editor as an input field in a dialog box, then consider using
+[`LanguageTextField`](upsource:///platform/platform-impl/src/com/intellij/ui/LanguageTextField.java),
+it provides a more Kotlin UI DSL friendly way of creating with the ability to set a default value on creation.
+
+If you want to add autocompletion to the editor, then consider using
+[`TextFieldWithCompletion`](upsource:///platform/platform-impl/src/com/intellij/util/textCompletion/TextFieldWithCompletion.java).
+The constructor takes as an argument a class that implements
+[`TextCompletionProvider`](upsource:///platform/platform-impl/src/com/intellij/util/textCompletion/TextCompletionProvider.java)
+to provide autocompletion variants.
+Use a class
+[`TextFieldCompletionProvider`](upsource:///platform/lang-impl/src/com/intellij/util/TextFieldCompletionProvider.java)
+to create your own provider. For this, override the `addCompletionVariants()` method in which put all the completion variants using the `CompletionResultSet.addElement()` method.
+
```

Comment:
worth mentioning TextFieldCompletionProviderDumbAware to allow completion even during indexing

Reply:
Added.